Autopilot/PathManager/Src/pathModeSelector.cpp

```diff
@@ -15,21 +15,69 @@ PathModeSelector* PathModeSelector::getInstance() {
 
 PathModeSelector::PathModeSelector() : current_mode_enum {MODE_TESTING_NONE} {
     current_mode = nullptr; 
+    first_instr = nullptr;
 }
 
 #else
 
 PathModeSelector::PathModeSelector() : current_mode_enum {MODE_TAKEOFF} {
     current_mode = &TakeoffMode::getInstance(); 
+    first_instr = nullptr;
 }
 
 #endif
 
 void PathModeSelector::execute(Telemetry_PIGO_t telemetry_in, SFOutput_t sensor_fusion_in, IMU_Data_t imu_data_in) {
+    
+    //Check whether the telemetry_in has an editflightPath command. If so, we enqueue it into our InstructionQueue
+    if (telemetry_in.waypointModifyFlightPathCommand != NO_FLIGHT_PATH_EDIT) {        
+        PathModeSelector::enqueueInstruction(telemetry_in);
```

Comment:
no need to write `PathModeSelector::`. Since you are actually in a method that is a part of the `PathModeSelector` class, you can access all public and private methods within it.

Also, we really only use `ClassName::` when we are calling static methods from functions outside of the ClassName class. You can read more about it here: https://stackoverflow.com/questions/4365982/how-do-i-call-a-static-method-of-another-class